File: memtest2.pl
#!/usr/bin/perl -w
use strict;

my $INNER = 10000;
my $LOOPS = 1*1000*1000;

sleep 1;

my %aa;
for (my $i = 0; $i < $LOOPS; $i++) {
  my $a = ' ' x $INNER;
  $aa{$i % 1000} = $a;
}

File: memtests/MemTest2.java
public class MemTest2
{
  static int INNER = 10000;
  static int LOOPS = 1*1000*1000;
  
  public static void main(String[] args)
  {
    try { Thread.sleep(1000); } catch (InterruptedException e) {}
    
    byte[][] aa = new byte[1000][];
    for (int i = 0; i < LOOPS; i++) {
      byte[] a = new byte[INNER];
      aa[i % 1000] = a;
    }
  }
}
